### DISCUSSION TITLE
PKCE Support?

### DISCUSSION THREAD

**Discussion by donavon** [sergiodxa/remix-auth-oauth2#9] [Q&A] [answered]
@sergiodxa, 

Does `remix-auth` support [PKCE](https://oauth.net/2/pkce/)? If not, are there plans to do so? Any auth solution without PKCE is considered a non-starter by some organizations, especially when building something like an Electron app, but even with fully server-side auth.

Thx,
Donavon

**Comment by sergiodxa** [comment]
@donavon the OAuth2Strategy set the state param on the URL and session and checks if they're same but the `code_challenge` and `code_challenge_method` are not implemented.

I'm moving this discussion to the remix-auth-oauth2 to keep track of this there.

**Comment by Djacob-no** [comment]
I agree to add this, to make it up to date with oauth 2.1 standard

**Comment by sergiodxa** [comment, marked as answer]
Support for this was implemented in #89
